Refresh.GameServer/Types/Assets/GameAsset.cs

```diff
@@ -23,4 +32,341 @@ public partial class GameAsset : IRealmObject
+
+        bool mainlineDoesntNeedConversion = this.AssetType is GameAssetType.Png or GameAssetType.Texture or GameAssetType.GameDataTexture;
+        
+        switch (game)
+        {
+            case TokenGame.Website:
+            case TokenGame.LittleBigPlanet1: 
+            case TokenGame.LittleBigPlanet2:
+            case TokenGame.LittleBigPlanet3:
+            case TokenGame.LittleBigPlanetVita: {
+                Image sourceImage = decodeImage(dataStorePath);
+                
+                //Load the image from the data store and transform it
+                Image? image = transformImage(sourceImage);
+                //If its null, then no transformation was needed
+                if (image == null)
+                {
+                    if (mainlineDoesntNeedConversion)
+                    {
+                        //Return the existing asset hash
+                        return this.AssetHash;
+                    }
+
+                    //Set the image to use to the source image
+                    image = sourceImage;
+                }
+
+                //Save the image as a PNG file in a byte array in memory
+                MemoryStream memory = new();
+                image.SaveAsPng(memory);
+                byte[] data = memory.ToArray();
+
+                //Get the hash of the converted asset
+                string convertedHash = AssetImporter.BytesToHexString(SHA1.HashData(data));
+
+                //Write the data to the store
+                dataStore.WriteToStore(convertedHash, data);
+
+                //Return the new icon hash
+                return convertedHash;
+            }
+            case TokenGame.LittleBigPlanetPSP: {
+                Image<Rgba32> sourceImage = decodeImage(dataStorePath);
+
+                //Transform the image, if no transformation is needed, use the source image
+                Image<Rgba32> image = transformImage(sourceImage) ?? sourceImage;
+
+                MemoryStream memory = new();
+                new MipEncoder().Encode(image, memory);
+                //Get the used chunk of the underlying buffer
+                Span<byte> data = memory.GetBuffer().AsSpan()[..((int)memory.Length)];
+                //Encrypt the data
+                byte[] encryptedData = ResourceHelper.PspEncrypt(data, Importer.PSPKey.Value);
+
+                //Get the hash
+                string convertedHash = AssetImporter.BytesToHexString(SHA1.HashData(encryptedData));
+
+                dataStore.WriteToStore($"psp/{convertedHash}", encryptedData);
+
+                return convertedHash;
+            }
+            default:
+                throw new ArgumentOutOfRangeException(nameof(game), game, null);
+        }
+    }
+
+    public string GetAsPhoto(TokenGame game, GameDatabaseContext database, IDataStore dataStore)
+    {
+        return this.GetAsGeneric(
+            game,
+            database,
+            dataStore,
+            this.PhotoTransformation,
```

Comment:
See my comment above

```suggestion
            () => null,
```